Bottom: 077b55694d09ea9242469b56fc4bbfb43fda75c7
Top:    f7e7bf715c63ddf0acefd63c0b14779d04aae50b
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-18 19:52:47 +0000

Refresh of limit-virtual-memory-size-of

---

diff --git a/src/polygon.c b/src/polygon.c
index ff7e60a..6d5dd4d 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -104,6 +104,10 @@ dicer output is used for HIDs which cannot render things with holes
 #include <dmalloc.h>
 #endif
 
+/* For getrlimit, setrlimit */
+#include <sys/time.h>
+#include <sys/resource.h>
+
 #define ROUND(x) ((long)(((x) >= 0 ? (x) + 0.5  : (x) - 0.5)))
 
 #define UNSUBTRACT_BLOAT 10
@@ -115,11 +119,18 @@ static double rotate_circle_seg[4];
 void
 polygon_init (void)
 {
+  struct rlimit limit;
+
   double cos_ang = cos (2.0 * M_PI / POLY_CIRC_SEGS_F);
   double sin_ang = sin (2.0 * M_PI / POLY_CIRC_SEGS_F);
 
   rotate_circle_seg[0] = cos_ang;  rotate_circle_seg[1] = -sin_ang;
   rotate_circle_seg[2] = sin_ang;  rotate_circle_seg[3] =  cos_ang;
+
+  /* DEBUG - AVOID PCB running the system out of memory! */
+  getrlimit (RLIMIT_AS, &limit);
+  limit.rlim_cur = MIN (limit.rlim_cur, 2000 * 1024 * 1024 /* 2 GiB limit to virtual memory size */);
+  setrlimit (RLIMIT_AS, &limit);
 }
 
 Cardinal
